openapi.yaml

```diff
@@ -0,0 +1,404 @@
+swagger: "2.0"
+info:
+  description: "reader api service"
+  version: "0.0.1"
+  title: "Reader API"
+  contact:
+    email: "hello@gorillab.co"
+host: "glab-reader-api.herokuapp.com"
+basePath: "/api"
+tags:
+- name: "auth"
+  description: "Auth module"
+- name: "user"
+  description: "User module"
+- name: "source"
+  description: "Source module"
+- name: "post"
+  description: "Post module"
+schemes:
+- "http"
+consumes:
+  - "application/json"
+produces:
+  - "application/json"
+paths:
+  /auth/facebook:
+    get:
+      tags:
+      - "auth"
+      summary: "Logs user into the system by facebook"
+      description: ""
+      operationId: ""
+      responses:
+        200:
+          description: "successful operation, send a facebook login form"
+  /auth/facebook/callback:
+    get:
+      tags:
+      - "auth"
+      summary: "Facebook call this api to return result of authentication"
+      description: ""
+      operationId: ""
+      responses:
+        200:
+          description: "successful operation"
+  /auth/logout:
+    get:
+      tags:
+      - "auth"
+      summary: "Logs out current logged in user session"
+      description: ""
+      operationId: "logout"
+      parameters: []
+      responses:
+        default:
+          description: "successful operation"
```

Comment:
all path responses should have 2 cases:
- 200 or another success status codes
- and default: an error case. should provide the error response in the definitions and this refers to it.